core/modules/keyboard.js

```diff
@@ -322,8 +322,21 @@ KeyboardManager.prototype.updateShortcutLists = function(tiddlerList) {
 };
 
 KeyboardManager.prototype.handleKeydownEvent = function(event) {
+	return this.handleKeydownEventInternal(event, false);
+}
+
+KeyboardManager.prototype.handlePrimaryKeydownEvent = function(event) {
+	return this.handleKeydownEventInternal(event, true);
+}
+
+KeyboardManager.prototype.handleKeydownEventInternal = function(event, onlyPrimary) {
 	var key, action;
 	for(var i=0; i<this.shortcutTiddlers.length; i++) {
+		var tiddler = $tw.wiki.getTiddler(this.shortcutTiddlers[i]);
```

Comment:
In line 313 there is `updateShortcutLists` .. which seems to copy info from the tiddler store to the widget state vars. So the execution has to be only done if the list changes. 

It seems here you access the tiddler store with every key-down event, which may have performance implications. ... I didn't run real tests. Only guessing by reading the code.

Reply:
It's one of those cases where it should only be a tiny performance cost but it's there and can be cached away since the cache mechanism is already in place - so it's cached now!